### DISCUSSION TITLE
run code when element is displayed

### DISCUSSION THREAD

**Discussion by guillaumeAbel** [slint-ui/slint#4719] [Q&A]
Hi everyone 👋 ,

I'm using slint with rust,

I'm doing some benchmarks so i have to run code when a slint element is displayed.

When i call self::show() on my element, it's not really displayed (only a rectangle of his shape when i click), and i have to call run_event_loop() to see it.

And obviously event_loop is a loop so run code after called it make no sense.

So i would like to call on_init() on my element, which is defined like: 

```slint
export component App inherits Window {
    width: 900px;
    height: 900px;
    init => {
    }
    StandardTableView {
        width: 100%;
        height: 100%;
        //columns and row definition
    }
}
```

I defined an empty init callback to be sure even if the doc say its by default.
This result to : 
```sh
app.on_init(|app| {
 |---^^^^^^^ method not found in `App`
```

I also tried like :
```slint
export component App inherits Window {
    width: 900px;
    height: 900px;
    table:= StandardTableView {
      init => {
      }
      //...
    }
```

But i cant access to app.table :
```sh
app.table.on_init(|app| {
 |      ^^^^^ unknown field
```

So if you know how to do this or if you have another way to run code when my element is displayed, Thanks You 👍  !

Thanks for reading !

**Comment by guillaumeAbel** [comment]
update:

I attempted to use a callback by invoking app.window().set_rendering_notifier() and monitoring the AfterRendering state in my callback function. 

However, the measured time is significantly shorter than the actual rendering time. 

There is a delay of several seconds before I can visualize my element, yet my callback only receives the AfterRendering state after 100 milliseconds.

**Comment by ogoffart** [comment]
you may try to make a new callback:

```
component Foo {
   callback created;
   init => { created(); }
}
```

And that created callback you can set an handler in Rust.
